apps/etterlatte-behandling/src/main/kotlin/grunnlagsendring/GrunnlagsendringshendelseService.kt

```diff
@@ -333,6 +334,30 @@ class GrunnlagsendringshendelseService(
             }.map { it.first }
     }
 
+    fun opprettDoedshendelseForPerson(grunnlagsendringshendelse: Grunnlagsendringshendelse): OppgaveIntern? {
+        val eksisterendeHendelse =
+            grunnlagsendringshendelseDao.hentGrunnlagsendringshendelserMedStatuserISak(
+                grunnlagsendringshendelse.sakId,
+                listOf(GrunnlagsendringStatus.VENTER_PAA_JOBB, GrunnlagsendringStatus.SJEKKET_AV_JOBB),
+            ).firstOrNull {
+                it.gjelderPerson == grunnlagsendringshendelse.gjelderPerson && it.type == grunnlagsendringshendelse.type
+            }
+
+        // Vi bryr oss ikke om det samsvarer eller ikke, men underliggende tjenester krever det.
+        val samsvarMellomKildeOgGrunnlag = SamsvarMellomKildeOgGrunnlag.Doedsdatoforhold(null, null, false)
+        if (eksisterendeHendelse != null) {
+            logger.info(
+                "Det finens allede en grunnlagsendringshendelse ${grunnlagsendringshendelse.type} med id " +
+                    "${grunnlagsendringshendelse.id} fra før, forsøker å returnere relevant oppgave.",
+            )
+            return oppgaveService.hentOppgaverForSak(grunnlagsendringshendelse.sakId)
+                .find { it.referanse == eksisterendeHendelse.id.toString() }
+        } else {
+            grunnlagsendringshendelseDao.opprettGrunnlagsendringshendelse(grunnlagsendringshendelse)
+            return oppdaterGrunnlagsEndringOgOpprettOppgave(grunnlagsendringshendelse, samsvarMellomKildeOgGrunnlag)
```

Comment:
Tenker vi her kun gjør en `grunnlagsendringshendelseDao.opprettGrunnlagsendringshendelse(grunnlagsendringshendelse)` med riktig state som gjøres i `oppdaterGrunnlagsEndringOgOpprettOppgave` + en `oppgaveService.opprettNyOppgaveMedSakOgReferanse` .

Reply:
Du tenker for å unngå å trenge en "oppdater"?

Reply:
ja

Reply:
[a71708a](https://github.com/navikt/pensjon-etterlatte-saksbehandling/pull/3927/commits/a71708a94e4e7b89572f6d255fc633ad6e601c68)